userspace/libsinsp/filterchecks.cpp

```diff
@@ -1873,19 +1873,19 @@ bool sinsp_filter_check_fd::compare(sinsp_evt *evt)
-	{PT_CHARBUF, EPF_NONE, PF_NA, "proc.args", "Arguments", "The arguments passed on the command line when starting the process generating the event excluding argv[0]."},
+	{PT_CHARBUF, EPF_NONE, PF_NA, "proc.exe", "First Argument", "The first command line argument argv[0] which is usually the executable name but it could be also a custom string, it depends on what the user specifies. This field could be populated through a procfs scan (specifically by reading from /proc/<pid>/cmdline) or through the syscall events stream at runtime (in this case, the field is read from the syscalls args)."},
+	{PT_CHARBUF, EPF_NONE, PF_NA, "proc.pexe", "Parent First Argument", "The proc.exe (first command line argument argv[0]) of the parent process."},
+	{PT_CHARBUF, EPF_NONE, PF_NA, "proc.aexe", "Ancestor First Argument", "The proc.exe (first command line argument argv[0]) for a specific process ancestor. You can access different levels of ancestors by using indices. For example, proc.aexe[1] retrieves the proc.exe of the parent process, proc.aexe[2] retrieves the proc.exe of the grandparent process, and so on. The current process's proc.exe line can be obtained using proc.aexe[0]. When used without any arguments, proc.aexe is applicable only in filters and matches any of the process ancestors. For instance, you can use `proc.aexe endswith java` to match any process ancestor whose command line contains the term `java`."},
+	{PT_CHARBUF, EPF_NONE, PF_NA, "proc.exepath", "Process Executable Path", "The full executable path of the process. This field could be populated through a procfs scan (in this case the field is the resolved path of /proc/<pid>/exe) or through the syscall events stream at runtime (in this case, the field is read from the syscalls args)."},
+	{PT_CHARBUF, EPF_NONE, PF_NA, "proc.pexepath", "Parent Process Executable Path", "The proc.exepath (full executable path) of the parent process."},
+	{PT_CHARBUF, EPF_NONE, PF_NA, "proc.aexepath", "Ancestor Executable Path", "The proc.exepath (full executable path) for a specific process ancestor. You can access different levels of ancestors by using indices. For example, proc.aexepath[1] retrieves the proc.exepath of the parent process, proc.aexepath[2] retrieves the proc.exepath of the grandparent process, and so on. The current process's proc.exepath line can be obtained using proc.aexepath[0]. When used without any arguments, proc.aexepath is applicable only in filters and matches any of the process ancestors. For instance, you can use `proc.aexepath endswith java` to match any process ancestor whose path ends with the term `java`."},
+	{PT_CHARBUF, EPF_NONE, PF_NA, "proc.name", "Name", "The name (excluding the path) of the executable generating the event (task->comm). This field could be populated through a procfs scan (in this case the field is read from /proc/<pid>/status) or through the syscall events stream at runtime (in this case the field is read from the syscalls args)."},
```

Comment:
AFAIK `task->comm` is truncated to 15 characters (however, the characters limit depends on a kernel setting).

Still determining if this also applies when the value is being collected through a procfs scan  (it should be so, @therealbobo, can you confirm?)

Anyway, this should be documented here since it may affect users trying to write a rule's condition.
Likewise, we should address this also for other fields with similar special cases that may not be self-evident, if any (ie. it's not obvious that one knows that `task->comm` is being truncated by the kernel).

@Andreagit97 wdyt?

Reply:
Given that the name in /proc/<pid>/status reflects `task->comm`, the name will be consistent in both cases (Procfs scan and syscall event). I'll add the truncation detail.

Reply:
Big plus one on the character limit, isn't it 16 characters? Let's check.

In addition @therealbobo since we are here could we elaborate that the last portion of the executable path and the proc.name (task->comm) can be different when for example a custom process name is specified which is very common for example for java processes.

Perhaps we should change the entire old phrasing `The name (excluding the path) of the executable` -> `The name of the process` throughout for all instances ... as referencing executable path here is misleading, see above comment?

Reply:
yep the length of `task->comm` is always 16 characters (`TASK_COMM_LEN = 16`)

Reply:
I said "truncated after 15 characters" because the 16th is the null byte 😄

Reply:
yep but since the user could manipulate it with prctl I'm not sure the last byte will be always `\0`. I would say 16 just to be sure